samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -502,6 +504,12 @@ void main(void)
 		error();
 	}
 
+	err = thingy53_dfu_init();
```

Comment:
Maybe I could add this under `IS_ENABLED(CONFIG_BOARD_THINGY53...)` ? That would be less confusing.

Reply:
It would be great to let user run our exiting samples (or at least some of them) on Thingy:53 in an easy way. With current implementation of SMP in Zephyr we actually need to register bt_smp and provide img_mgmt handlers somewhere in source code. We also need to add code that enables Bluetooth and establishes connection. So far, we were thinking about integrating this in sample/application (since there is no way to do it automagically in the background).

Reply:
We also need to make sure that we use the same FLASH partitions configuration and proper configuration of MCUBoot bootloader (partition sizes and signing key must be consistent; serial recovery needs to be enabled too in case application fails). These configuration files are specific to NCS build system so we cannot add them under Zephyr's board directory.

Reply:
Using a separate set of samples dedicated for Thigny:53 could be considered, but we would need to maintain samples that are dedicated for particular board which is not a good approach in my opinion.

Reply:
@MarekPieta please create a weak function `board_init` in main.c. Call it from this place - no comment is needed as name is self defining.
e.g.
```
__weak void board_init(void)
{
   /* Nothing specific is required for a generic board. */
}
```

Reply:
Then in per board file simply override the weak function. I would even ignore header.

Reply:
Do not mention thingy53 anywhere outside of a single board related file.

Reply:
Fixed